server/controllers/finance/fiscal.js

```diff
@@ -457,3 +459,218 @@ function notNullBalance(array) {
     return (item.debit !== 0 || item.credit !== 0);
   });
 }
+
+/**
+ * @function closing
+ * @description closing a fiscal year
+ */
+function closing(req, res, next) {
+  let id = req.params.id,
```

Comment:
Just for you information, this is the problem `let` was trying to solve.  With `var`, all variables are implicitly declared at the top due to [variable hoisting](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/var), so JS programmers would write `var id, x, y, z;` at the top of their scopes to show that those variables are defined an available.

With `let`, the declarations are not hoisted, leaving a [temporal dead zone](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/let).  So, you no longer have to put all your `let` declarations at the top, since they will only be defined once you type `let variable = value;`.

As a concrete example, here is a behavior to try in the console;
```js
function temp() {
   console.log('i', i);
   var i = 0;
   console.log('i', i);
}

temp(); /// logs 'i' 'undefined', then 'i' 0.

function temp2() {
  console.log('x', x);
  let x = 0;
  console.log('x', 0);
}

temp2(); // ReferenceError: x is not defined
```

For this reason, it is best practice to only declare variables with `let` and `const` where they occur.  It solves one of JavaScript's weaknesses.